src/main/java/seedu/address/logic/CommandList.java

```diff
@@ -2,60 +2,86 @@
 
 import java.util.ArrayList;
 
+import seedu.address.logic.commands.NextCommand;
+import seedu.address.logic.commands.PreviousCommand;
+import seedu.address.logic.commands.exceptions.HistoryException;
+
+
 /**
  * Records the latest user inputs.
  * Only the latest MAX_RECORD_NUMBER inputs will be recorded.
  */
 public class CommandList {
-    private static final int MAX_RECORD_NUMBER = 3;
-    private static final ArrayList<String> inputHistory = new ArrayList<>();
+    private static CommandList theList = null;
+    private static final int MAX_RECORD_NUMBER = 10;
+    private int pointer = -1;
+    private final ArrayList<String> inputHistory = new ArrayList<>();
 
+    private CommandList() {}
     /**
      * Records the latest user input.
      * @param lastCommand the last user input.
      */
-    public static void record(String lastCommand) {
-        if (!lastCommand.equals("history")) {
-            inputHistory.add(lastCommand);
+    public void record(String lastCommand) {
+        if (lastCommand.equalsIgnoreCase(PreviousCommand.COMMAND_WORD)) {
+            return;
         }
+        if (lastCommand.equalsIgnoreCase(NextCommand.COMMAND_WORD)) {
+            return;
+        }
+        inputHistory.add(lastCommand);
         if (inputHistory.size() > MAX_RECORD_NUMBER) {
             clearOldCommand();
+            return;
         }
+        pointer++;
     }
 
     /** Gets the latest user input. */
-    public static String getLastCommand() {
-        if (isEmpty()) {
-            return "";
+    public String getLastCommand() {
```

Comment:
Shouldn't this be renamed to getCurrentCommand() or something similar? Since it doesn't get the last command anymore